src/test-utils.tsx

```diff
@@ -50,6 +51,23 @@ const customRender = (ui: ReactElement) => render(ui, { wrapper: WithProviders }
 const customRenderHook = <Result, Props>(hook: (initialProps: Props) => Result) =>
   renderHook(hook, { wrapper: WithProviders })
 
+/**
+ * Casts the passed function as a jest.Mock.
+ * Use this in combination with jest.mock() to safely access function from mocked modules.
+ *
+ * @example
+ * import { useExample } from 'example'
+ * jest.mock('example', () => ({ useExample: jest.fn() }))
+ * beforeEach(() => {
+ *   asMock(useExample).mockImplementation(() => ...)
+ * })
+ */
+export function asMock<T extends (...args: any) => any>(fn: T) {
```

Comment:
are you sure this is the only way to do this? i am surprised that if this is such a common use case, jest wouldn't include an easier to way to do this

Reply:
I have not found anything else, and I've looked for it many times (because it is a common use case). I _can_ improve it though, using https://jestjs.io/docs/jest-object#jestismockfunctionfn.

Reply:
This has been broken into its own PR: https://github.com/Uniswap/interface/pull/6310.